### PR TITLE
feat: added editing cards tests

### PR DESCRIPTION
**Pull request by wawrzonnn**

### CODE REVIEW

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

The code changes in `__tests__/cardRoutes.test.ts` file cover the task requirements as follows:

1. **Test that the function returns a status code of 200.**

The test `'should return status code 200 when updating a card'` checks if the status code returned by the function is 200. This test case meets the first requirement.

```javascript
it('should return status code 200 when updating a card', async () => {
    // ...
    expect(response.status).toBe(200);
});
```

2. **Test that the function updates the requested flashcard with the correct fields.**

The test `'should update the card with the correct fields'` checks if the function updates the flashcard with the correct fields. This test case meets the second requirement.

```javascript
it('should update the card with the correct fields', async () => {
    // ...
    expect(updatedCard!.front).toBe(updatedData.front);
    expect(updatedCard!.back).toBe(updatedData.back);
    expect(updatedCard!.tags).toEqual(expect.arrayContaining(updatedData.tags));
});
```

3. **Test that the function returns the updated flashcard.**

The test `'should return the updated flashcard'` checks if the function returns the updated flashcard. This test case meets the third requirement.

```javascript
it('should return the updated flashcard', async () => {
    // ...
    expect(response.body.front).toBe(updatedData.front);
    expect(response.body.back).toBe(updatedData.back);
    expect(response.body.tags).toEqual(expect.arrayContaining(updatedData.tags));
});
```

In conclusion, all task requirements are covered by the provided code changes.
